ShareX.ScreenCaptureLib/Shapes/Tool/CutOutTool.cs

```diff
@@ -0,0 +1,188 @@
+                    bool overflowsLeft = buttonsLeft < Manager.Form.ClientArea.Left + buttonOffset;
+                    bool overflowsRight = buttonsRight >= Manager.Form.ClientArea.Right - buttonOffset;
+                    if (overflowsLeft && overflowsRight)
+                    {
+                        // can't fix
+                    }
+                    else if (overflowsLeft)
+                    {
+                        buttonsLeft = Manager.Form.ClientArea.Left + buttonOffset;
+                    }
+                    else if (overflowsRight)
+                    {
+                        buttonsRight = Manager.Form.ClientArea.Right - buttonOffset;
+                        buttonsLeft = buttonsRight - 2 * buttonSize.Width - buttonOffset;
+                    }
+                    confirmButton.Rectangle = new RectangleF(buttonsLeft, buttonsTop, buttonSize.Width, buttonSize.Height);
+                    cancelButton.Rectangle = confirmButton.Rectangle.LocationOffset(buttonSize.Width + buttonOffset, 0);
+                }
+                else
+                {
+                    float spaceRight = Manager.Form.ClientArea.Right - Rectangle.Right;
+                    bool positionRight = spaceRight >= buttonSize.Width + 2 * buttonOffset;
+                    float buttonsLeft = positionRight ? Rectangle.Right + buttonOffset : Rectangle.Left - buttonOffset - buttonSize.Width;
+                    float buttonsTop = Rectangle.Top + Rectangle.Height / 2 - (2 * buttonSize.Height + buttonOffset) / 2;
+                    float buttonsBottom = buttonsTop + 2 * buttonSize.Height + buttonOffset;
+                    bool overflowsTop = buttonsTop < Manager.Form.ClientArea.Top + buttonOffset;
+                    bool overflowsBottom = buttonsBottom >= Manager.Form.ClientArea.Bottom - buttonOffset;
+                    if (overflowsTop && overflowsBottom)
+                    {
+                        // can't fix
+                    }
+                    else if (overflowsTop)
+                    {
+                        buttonsTop = Manager.Form.ClientArea.Top + buttonOffset;
+                    }
+                    else if (overflowsBottom)
+                    {
+                        buttonsBottom = Manager.Form.ClientArea.Bottom - buttonOffset;
+                        buttonsTop = buttonsBottom - 2 * buttonSize.Height - buttonOffset;
+                    }
+                    confirmButton.Rectangle = new RectangleF(buttonsLeft, buttonsTop, buttonSize.Width, buttonSize.Height);
+                    cancelButton.Rectangle = confirmButton.Rectangle.LocationOffset(0, buttonSize.Height + buttonOffset);
+                }
+            }
+        }
+
+        public override void OnDraw(Graphics g)
+        {
+            if (IsHorizontalTrim)
+            {
+                g.FillRectangle(EffectBrush, new RectangleF(Rectangle.X, g.ClipBounds.Y, Rectangle.Width, g.ClipBounds.Height));
```

Comment:
I wonder how it would look if we used transparent checker pattern using texture brush, like this:

![](https://jaex.getsharex.com/2022/08/devenv_jCakUO2bOD.png)

Because currently in gray background it can be hard to see selection, because selection is also gray.

Reply:
I'm not entirely sold on it, but on the other hand maybe it does better denote "this part is getting tossed out".

![image](https://user-images.githubusercontent.com/1062071/185166400-783d1fcd-ff88-40bc-abdf-b890fcff62e6.png)

Reply:
I mean't like this:

```
using (Image selectionHighlightPattern = ImageHelpers.CreateCheckerPattern(1, 1, Color.FromArgb(128, Color.LightGray), Color.FromArgb(128, Color.Gray)))
```

![](https://jaex.getsharex.com/2022/08/ShareX_3HNxBZ09nd.png)